lib/drum.hoon

```diff
@@ -698,10 +698,115 @@
     =.  +>  (ta-dog(say.inp (~(commit sole say.inp) ted)) ted)
     +>
   ::
+  ++  alnum
+    |=  a/@  ^-  ?
+    ?|  &((gte a '0') (lte a '9'))
+        &((gte a 'A') (lte a 'Z'))
+        &((gte a 'a') (lte a 'z'))
+    ==
+  ::
+  ++  next-edge
+    |=  buf/(list @c)
+    =|  i/@ud
+    =+  m=|
+    |-  ^+  i
+    ?~  buf  i
+    =+  w=(alnum i.buf)
+    =.  m  |(m w)
+    ?:  &(m !|(=(0 i) w))
+      i
+    $(i +(i), buf t.buf)
+  ::
+  ++  next-word
+    |=  buf/(list @c)
+    =|  i/@ud
+    |-  ^+  i
+    ?:  |(?=($~ buf) (alnum i.buf))
+      i
+    $(i +(i), buf t.buf)
+  ::
+  ++  jump-bwrd
+    |=  {pos/@ud buf/(list @c)}
+    ^-  @ud
+    (sub pos (next-edge (flop (scag pos buf))))
+  ::
+  ++  jump-fwrd
+    |=  {pos/@ud buf/(list @c)}
+    ^-  @ud
+    (add pos (next-edge (slag pos buf)))
+  ::
   ++  ta-met                                          ::  meta key
     |=  key/@ud
-    ~&  [%ta-met key]
-    +>
+    ?+    key    ta-bel
+      $dot  ?.  &(?=(^ old.hit) ?=(^ -.old.hit))
+              ta-bel
+            =+  old=`(list @c)`-.old.hit
+            =+  b=(jump-bwrd (dec (lent old)) old)
+            %-  ta-hom(ris ~)
+            (ta-cat pos.inp (slag b old))
+            ::
+      $bac  ?:  =(0 pos.inp)
+              ta-bel
+            =+  b=(jump-bwrd pos.inp buf.say.inp)
+            %-  ta-hom(kil `(slag b (scag pos.inp buf.say.inp)), ris ~)
+            (ta-cut b (sub pos.inp b))
+            ::
+      $b    ?:  =(0 pos.inp)
+              ta-bel
+            +>(pos.inp (jump-bwrd pos.inp buf.say.inp))
+            ::
+      $c    ?:  =(pos.inp (lent buf.say.inp))
+              ta-bel
+            =+  sop=(add pos.inp (next-word (slag pos.inp buf.say.inp)))
+            =+  ^=  prod
+            %-  ta-hom
+            :~  %mor
+              [%del sop]
+              :+  %ins  sop
+              (turf (cuss [(tuft (snag sop buf.say.inp))]~))
+            ==
+            prod(pos.inp (jump-fwrd sop buf.say.inp))
+            ::
```

Comment:
This is also awkward, but pretty much required... It's an odd shortcut: a single char edit, then a positional jump.